OpenInFocus/ActionViewController.swift

```diff
@@ -44,24 +44,50 @@ class ActionViewController: SLComposeServiceViewController {
         return NSURL(string: "\(self.urlScheme)://open-url?url=\(url)")
     }
 
+    func textUrl(text: String) -> NSURL? {
+        guard let query = text.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else { return nil }
+        return NSURL(string: "\(self.urlScheme)://open-text?text=\(query)")
+    }
+
     override func configurationItems() -> [Any]! {
-        guard let item = extensionContext?.inputItems.first.flatMap({ $0 as? NSExtensionItem }) else { fatalError() }
-        guard let itemProvider = item.attachments?.first.flatMap({ $0 as? NSItemProvider }) else { fatalError() }
-
-        if itemProvider.isText {
-            itemProvider.processText { (textItem, error) in
-                let text = textItem as? String
-                let url = text.flatMap(NSURL.init(string:))
-                guard let focusUrl = url.flatMap({ $0.encodedUrl }).flatMap(self.focusUrl) else { self.cancel(); return }
-                self.handleUrl(focusUrl)
+        if let inputItems: [NSExtensionItem] = extensionContext?.inputItems as? [NSExtensionItem] {
+            var urlProvider: NSItemProvider?
+            var textProvider: NSItemProvider?
+
+            for item: NSExtensionItem in inputItems {
+                let attachments: [NSItemProvider] = (item.attachments as? [NSItemProvider]) ?? []
+                for attachment in attachments {
+                    if attachment.isUrl{
+                        urlProvider = attachment
+                        break;
```

Comment:
don't think you need the semi colon??

Reply:
Also, can something be text AND url? If so, the break makes sense

Reply:
@bkmunar yes, Feedly shares a title and a URL (Just like we do now) so in that case we only want the url.